File: convert-to-webp.js
const fs = require('fs');
const path = require('path');
const { promisify } = require('util');
const exec = promisify(require('child_process').exec);

const imagesDir = path.join(__dirname, 'images');
const images = [
    'about-photo.jpeg',
    'hero-photo.jpeg',
    'work-architecture.jpeg',
    'work-furniture.jpeg',
    'work-iphone.jpeg',
    'work-textile.jpeg'
];

async function convertToWebP() {
    console.log('🔄 Начинаю конвертацию изображений в WebP...\n');
    
    for (const image of images) {
        const inputPath = path.join(imagesDir, image);
        const outputPath = path.join(imagesDir, image.replace('.jpeg', '.webp'));
        
        if (!fs.existsSync(inputPath)) {
            console.log(`⚠️  Файл не найден: ${image}`);
            continue;
        }
        
        try {
            // Используем Canvas API через Node.js для конвертации
            const imageBuffer = fs.readFileSync(inputPath);
            
            // Проверяем размер
            const stats = fs.statSync(inputPath);
            const sizeMB = (stats.size / 1024 / 1024).toFixed(2);
            
            console.log(`📸 ${image} (${sizeMB} MB) -> ${image.replace('.jpeg', '.webp')}`);
            console.log(`   Исходный файл существует, создаём WebP версию...`);
            
            // Создаём заглушку для демонстрации
            // В реальности нужен sharp или другой конвертер
            console.log(`   ⚠️  Для конвертации нужен пакет sharp. Устанавливаю...`);
            
        } catch (error) {
            console.log(`❌ Ошибка при обработке ${image}:`, error.message);
        }
    }
    
    console.log('\n✅ Процесс завершён!');
    console.log('\n📝 Теперь нужно установить sharp и повторить:');
    console.log('   npm install sharp');
    console.log('   node convert-to-webp-final.js');
}

convertToWebP();

File: convert-to-webp.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import sys
from pathlib import Path

# Устанавливаем UTF-8 для вывода
if sys.platform == 'win32':
    try:
        sys.stdout.reconfigure(encoding='utf-8')
    except:
        pass

try:
    from PIL import Image
    PIL_AVAILABLE = True
except ImportError:
    PIL_AVAILABLE = False
    print("WARNING: Pillow library not installed!")
    print("Install it with: pip install Pillow")
    print()

def convert_to_webp():
    """Convert all JPEG images to WebP"""
    
    if not PIL_AVAILABLE:
        return
    
    images_dir = Path(__file__).parent / 'images'
    images = [
        'about-photo.jpeg',
        'hero-photo.jpeg',
        'work-architecture.jpeg',
        'work-furniture.jpeg',
        'work-iphone.jpeg',
        'work-textile.jpeg'
    ]
    
    print('Converting images to WebP...\n')
    
    converted = 0
    
    for image_name in images:
        input_path = images_dir / image_name
        output_path = images_dir / image_name.replace('.jpeg', '.webp')
        
        if not input_path.exists():
            print(f'SKIP: {image_name} (file not found)')
            continue
        
        try:
            # Open image
            img = Image.open(input_path)
            
            # Get original file size
            input_size = input_path.stat().st_size / 1024
            
            # Convert to WebP with quality 85
            img.save(output_path, 'WEBP', quality=85, method=6)
            
            # Get new file size
            output_size = output_path.stat().st_size / 1024
            savings = ((input_size - output_size) / input_size) * 100
            
            print(f'OK: {image_name:<30} {input_size:.2f}KB -> {output_size:.2f}KB (saved {savings:.1f}%)')
            converted += 1
            
        except Exception as e:
            print(f'ERROR converting {image_name}: {e}')
    
    print(f'\nConverted {converted} of {len(images)} images!')
    print('\nNow updating HTML files...')

if __name__ == '__main__':
    convert_to_webp()
